### PR TITLE
added pid files. permission denied error

### PR DESCRIPTION
**Pull request by ivcosla**

### CODE REVIEW

**Comment by ivcosla** [comment]
Closes #388

**Comment by ayuryshev** [comment]
So how this PR is intended to work?
What becomes if skywire-node panics with unrecovered error?
What will be on restart of this skywire-node?
Is there a recipe to detect hanged apps without node?

**Comment by ivcosla** [comment]
@ayuryshev 
1) When apps are spawned their PIDS are persisted in disk. 
2) When node is started, before it spawns apps, it reads such file and kills every app previously registered, then clears the pid file. Then apps are spawned and we go back to step 1).

If a node panics, when it is restarted it should kill the apps before it spawns the new ones, so app-node communication will work.

The only way to test it that I had was to run integration environments and use apps until one of them hangs on node close. You can check that by using `ps aux| grep [app name]`, then you can start the node again and check in the logs that the new app spawned correctly, if you perform the above command again you will see now such app has a different PID.

You can also check how every time that you start a node there is an `apps.pid` file inside `~/.skycoin/skywire/[node pk]/` directory with the name and pid of every node-spawned app. On every run it's pids must be different.

**Comment by ayuryshev** [comment]
I just today found a reproducible way to panic node leaving hanged  apps:

in skywire-services: 
1. run-messaging-env.sh
2. start messaging
3. kill messaging-server

I see that NodeA and NodeC panics 
and leave skychat running
must be good for testing this PR

File: pkg/node/node.go [hunk 1]
```diff
@@ -239,6 +249,61 @@ func (node *Node) Start() error {
 	return nil
 }
 
+func (node *Node) dir() string {
+	return pathutil.NodeDir(node.config.Node.StaticPubKey)
+}
+
+func (node *Node) pidFile() *os.File {
+	f, err := os.OpenFile(filepath.Join(node.dir(), "apps.pid"), os.O_RDWR|os.O_CREATE, 0600)
```
Comment (ayuryshev) [inline pkg/node/node.go]: 0644 maybe? To give ability to read for not-owners

File: pkg/node/node.go [hunk 2]
```diff
@@ -389,6 +459,14 @@ func (node *Node) SpawnApp(config *AppConfig, startCh chan<- struct{}) error {
 	return appErr
 }
 
+func (node *Node) persistPID(name string, pid int) {
```
Comment (ayuryshev) [inline pkg/node/node.go:462]: Do I understand correctly that content of pid-file is:
app-name pid

?

**Review by ayuryshev — COMMENTED** [review]
1. File permissions 0750 and 0600 (for directories and files) -  why not 755 and 0644 (typically used for pidfiles) ?
Examples:
- https://github.com/facebookarchive/pidfile/blob/f242e2999868dcd267a2b86e49ce1f9cf9e15b16/pidfile.go#L48
- https://github.com/moby/moby/blob/349d4dd0c981232a95fc178563a54792699d2823/pkg/pidfile/pidfile.go#L40

2. Did I understood correctly that one pid per node contains multiple lines one per application?

File: pkg/node/node.go (hunk 2 above)
Comment (ivcosla) [reply pkg/node/node.go:462]: yes

File: pkg/node/node.go (hunk 1 above)
Comment (ivcosla) [reply pkg/node/node.go]: it was 0644 before, but go-sec linter complained about it and pointed me to use 0600 instead.

Comment (evanlinjin) [reply pkg/node/node.go]: lol, guess 0600 is safer anyhow hahaha.

**Review by evanlinjin — APPROVED** [review]
Looks good.

**Pull request merged**